.github/workflows/merged.yml

```diff
@@ -0,0 +1,44 @@
+name: push main
+
+on:
+  pull_request:
+    branches:
+      - main
+    types: [closed]
+
+jobs:
+  update_version:
+    runs-on: ubuntu-20.04
+    if: github.event.pull_request.merged == true
+    timeout-minutes: 10
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: setup python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+
+      - name: install packages
+        run: >-
+          python -m pip install -r proto/requirements.txt
+
+      - name: protobuf build
+        run: |
+          cd proto
+          make
+
+      - name: update .proto_hash
+        run: |
+          cd proto
+          proto_hash=$(python3 proto_hash.py)
+          if [ ${proto_hash} != $(cat .proto_hash) ]; then
+            echo ${proto_hash} > .proto_hash
+            python3 increment_minor.py $(cat .version) > .version
+          fi
+          make clean
+
+      - uses: stefanzweifel/git-auto-commit-action@v3.0.0
+        with:
+          commit_message: update otaclient_pb2 hash and version
```

Comment:
@obi-t4 In this step, what files will be committed with this commit? I guess it implicitly commits files generated by the previous step, but I think explicitly specifying which files to be committed is more safe and preventing unintended files included.

P.S., the action provides the following option, maybe we can use this to specify which files to be included:
```
    # Optional glob pattern of files which should be added to the commit
    # Defaults to all (.)
    # See the `pathspec`-documentation for git
    # - https://git-scm.com/docs/git-add#Documentation/git-add.txt-ltpathspecgt82308203
    # - https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefpathspecapathspec
    file_pattern: '*.php src/*.js tests/*.js'
```

Reply:
@Bodong-Yang 
.proto_hash and .version will be committed if those are changed.
file_pattern didn't work properly. I tried --skip-error option.

Reply:
> @Bodong-Yang .proto_hash and .version will be committed if those are changed. file_pattern didn't work properly. I tried --skip-error option but it didn't work.

which file_pattern string did you use? It seems that this pattern is directly fed to `git add` command.

Reply:
The error like this:
https://github.com/obi-t4/ota-client/actions/runs/3358637312/jobs/5565860514
```
      - uses: stefanzweifel/git-auto-commit-action@v3.0.0
        with:
          commit_message: update otaclient_pb2 hash and version
          file_pattern: 'proto/.version proto/.proto_hash'
```
![image](https://user-images.githubusercontent.com/58019445/198932568-898ebda4-a0b4-4b9d-99b0-6fbfa0f4bef9.png)

Reply:
'proto/.version proto/.proto_hash' -> `'` should be removed.

Reply:
> 'proto/.version proto/.proto_hash' -> ' should be removed. 

removed, but it does't work.

Reply:
OK let me have a check.

Reply:
I tried the following:
```
$ git add 'proto/.proto_hash proto/.version'
warning: could not open directory 'proto/.proto_hash proto/': No such file or directory
fatal: pathspec 'proto/.proto_hash proto/.version' did not match any files
```

Reply:
@obi-t4 use `stefanzweifel/git-auto-commit-action@v4` instead, v4 fixed this bug.

```
Run stefanzweifel/git-auto-commit-action@v4
Started: bash /home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/entrypoint.sh
INPUT_REPOSITORY value: .
INPUT_STATUS_OPTIONS: 
INPUT_FILE_PATTERN: :/proto/.proto_hash :/proto/.version
INPUT_BRANCH value: test_ga
From https://github.com/Bodong-Yang/ota-client
 * [new branch]      backward_compat/grub_control -> origin/backward_compat/grub_control
 * [new branch]      main                 -> origin/main
 * [new branch]      ota-client-refactor3 -> origin/ota-client-refactor3
 * [new branch]      ota-rebuild-mode     -> origin/ota-rebuild-mode
 * [new branch]      otaclient_refactoring3_rc -> origin/otaclient_refactoring3_rc
 * [new branch]      refine/otaproxy      -> origin/refine/otaproxy
 * [new branch]      refine/reginf        -> origin/refine/reginf
 * [new branch]      test_ga              -> origin/test_ga
 * [new branch]      v[1](https://github.com/Bodong-Yang/ota-client/actions/runs/3358823481/jobs/5566240232#step:7:1)-legacy            -> origin/v1-legacy
Switched to a new branch 'test_ga'
M	proto/.proto_hash
M	proto/.version
branch 'test_ga' set up to track 'origin/test_ga'.
INPUT_ADD_OPTIONS: 
INPUT_FILE_PATTERN: :/proto/.proto_hash :/proto/.version
INPUT_COMMIT_OPTIONS: 
INPUT_COMMIT_USER_NAME: github-actions[bot]
INPUT_COMMIT_USER_EMAIL: github-actions[bot]@users.noreply.github.com
INPUT_COMMIT_MESSAGE: update otaclient_pb2 hash and version
INPUT_COMMIT_AUTHOR: Bodong-Yang <Bodong-Yang@users.noreply.github.com>
[test_ga f5c89dd] update otaclient_pb2 hash and version
 Author: Bodong-Yang <Bodong-Yang@users.noreply.github.com>
 2 files changed, 2 insertions(+), 2 deletions(-)
INPUT_TAGGING_MESSAGE: 
No tagging message supplied. No tag will be added.
INPUT_PUSH_OPTIONS: 
To https://github.com/Bodong-Yang/ota-client
   0cc12f1..f5c89dd  HEAD -> test_ga
branch 'test_ga' set up to track 'origin/test_ga'.
```

Also, this action push the commit to the branch where we submit the PR from, but this commit doesn't include into the branch we are merging to.

For the above example, I worked on branch `test_ga` and want to merge into `test_base`. After the PR is merged and closed, the action commits the `.proto_hash` and `.version` to `test_ga`, but this commit not being included in `test_base`.

Reply:
check https://github.com/Bodong-Yang/ota-client/commits/test_ga and https://github.com/Bodong-Yang/ota-client/commits/test_base for more details.

Reply:
> For the above example, I worked on branch test_ga and want to merge into test_base. After the PR is merged and closed, the action commits the .proto_hash and .version to test_ga, but this commit not being included in test_base.

Add one line that specifies which branch to commit to fix this problem:
```
      - uses: stefanzweifel/git-auto-commit-action@v4
        with:
          commit_message: update otaclient_pb2 hash and version
          file_pattern: ':/proto/.proto_hash :/proto/.version'
          branch: test_base
```

Reply:
@obi-t4 update the `merged.yml` as follow will solve the problem:
```yaml
      - uses: stefanzweifel/git-auto-commit-action@v4
        with:
          commit_message: update otaclient_pb2 hash and version
          file_pattern: ':/proto/.proto_hash :/proto/.version'
          branch: main
```
With the above change, now the `merged.yml` will only commit `proto/.proto_hash` and `proto/.version`, and this commit will come right after the PR merged commit in the main branch.

Reply:
@Bodong-Yang Thanks!
I'll handle how to specify branch to merge.

Reply:
Just `main` is enough.

Reply:
@Bodong-Yang 
fixed by: https://github.com/tier4/ota-client/pull/177/commits/812ecbc6e7f2ecd8ad50248dea14dacf54ee0a57